misc-tools/dj_make_chroot.in

```diff
@@ -396,6 +401,29 @@ if [ "$DISTRO" = 'Ubuntu' ]; then
     in_chroot "ln -s /bin/true /sbin/initctl"
 fi
 
+# Add additional repository lists to the chroot for upstream repos
+if [ -n "$REPOLISTS" ]; then
+    SPLIT_LIST=$(printf %s "$REPOLISTS" | tr ',' ' ')
+    for provided_repo in $SPLIT_LIST ; do
+        cat "$provided_repo" >> "$CHROOTDIR/etc/apt/sources.list.d/upstream_chroot.list"
+        for filetype in asc arm gpg; do
+            if [ -f "${provided_repo}.${filetype}" ]; then
+                GPGFILENAME=$(mktemp -p "$CHROOTDIR/etc/apt/trusted.gpg.d" 'upstream_gpg_XXXXXX.gpg')
+                file_head=$(head -n1 "${provided_repo}.${filetype}")
+                repo_sign_key="${provided_repo}.gpg"
+                if [ "-----BEGIN PGP ARMORED FILE-----" = "$file_head" ]; then
+                    gpg --dearmor "${provided_repo}.${filetype}"
+                    repo_sign_key="${provided_repo}.${filetype}.gpg"
```

Comment:
Nitpick: here I'd just use `$repo_sign_key` on the right-hand side, but this is fine.

Reply:
This is indeed wrong, but for another reason. I'll fix this in the train.
for the case pypy.asc we would need the file pypy.asc.gpg (aftor the dearmor .gpg is appended) or pypy.asc if its already in dearmored state (but would have the wrong filetype) but we would try to copy pypy.gpg which is wrong.

Reply:
Should now be fixed.